rskj-core/src/main/java/co/rsk/config/NodeCliOptions.java

```diff
@@ -41,6 +52,62 @@ public Config withConfig(Config config, String configValue) {
             return config.withValue(SystemProperties.PROPERTY_BASE_PATH, ConfigValueFactory.fromAnyRef(configValue));
         }
     },
+    SYNC_MODE("sync-mode", true) {
+        @Override
+        public Config withConfig(Config config, String configValue) {
+            if(configValue.equalsIgnoreCase("full") || configValue.equalsIgnoreCase("snap")){
+                return config.withValue(RskSystemProperties.PROPERTY_SYNC_MODE, ConfigValueFactory.fromAnyRef(configValue));
+            }else {
+                throw new IllegalArgumentException("Invalid sync mode: " + configValue + ". The valid options are <full> or <snap>.");
+            }
+        }
+    },
+    SNAP_NODES("snap-nodes", true) {
+        @Override
+        public Config withConfig(Config config, String configValue) {
+            try {
+                List<ConfigObject> snapConfigObjects = Arrays.stream(configValue.substring(1, configValue.length() - 1).split(","))
```

Comment:
could you pls explain why do we need to do `substring()` here?

Reply:
addressed, removed the substring. The flag should be formatted as --snap-nodes=<url1>`` --snap-nodes=<url2>, with each URL specified separately